tests/e2e/pipelinerollout_test.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+)
+
+var pipelineSpecSourceRPU = int64(5)
+var pipelineSpecSourceDuration = metav1.Duration{
+	Duration: time.Second,
+}
+var pipelineSpec = numaflowv1.PipelineSpec{
```

Comment:
Do we have plans in future to read spec from test file? Maybe we could load the entire PipelineRollout spec from file.

Reply:
Not yet, we can add an issue for that to add later. Numaflow and the previous Numaplane e2e tests did accommodate for this. It would be good to add in the future as now the tests are locked to these simple manifests.

Reply:
do you mind creating an issue to track this? Thanks!